### PR TITLE
ci: yamory job - cache is not stable

### PR DESCRIPTION
**Pull request by tuanphamcybozu**
## Why
This pull request fixes the pnpm cache which is not stable in yamory job. 

## What
Use actions/cache@v3 instead of 'cache' option in actions/setup-node@v3

## How to test
- Pass yamory job

## Checklist

- [x] Read [CONTRIBUTING.md](https://github.com/kintone/cli-kintone/blob/main/CONTRIBUTING.md)
- [ ] Updated documentation if it is required.
- [ ] Added tests if it is required.
- [x] Passed `pnpm lint` and `pnpm test` on the root directory.

### CODE REVIEW

File: .github/workflows/license.yml [hunk 1]
```diff
@@ -58,6 +59,12 @@ jobs:
...
           node-version: 16.x
 
+      - name: Set up pnpm cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.pnpm-store
+          key: ${{ runner.os }}-pnpm-${{ hashFiles('**/pnpm-lock.yaml') }}
+
       - name: Install pnpm
         run:  npm install -g pnpm@8.6.5
```
Comment (mshrtsr) [inline .github/workflows/license.yml:69]: Please use `pnpm/action-setup@v2` here.

**Review by tasshi-me — CHANGES_REQUESTED** [review]
@tuanphamcybozu 
Thank you for your PR!
I found that the only we have to do is disable the cache in the Yamory workflow.

---
The root cause of the error in Yamory workflow was it tried to cache node_modules.
In the Yamory workflow, we don't run `pnpm install`, so `node_modules` isn't generated.

As proof, all other workflows passed.
I checked the behavior with `pnpm install` step, and it passed.
https://github.com/kintone/cli-kintone/actions/runs/5583728474/jobs/10204331510

Yamory script (https://localscanner.yamory.io/script/pnpm) doesn't use the `node_modules` directory.
The `cache` option of setup-node is for storing `node_modules`.

In conclusion, the cache for the Yamory workflow isn't necessary, so we can remove the cache configuration for the Yamory workflow.

**Comment by tuanphamcybozu** [comment]
@mshrtsr 
Thanks for comment. I updated yamory job. 
Is it ok if we still keep change in other jobs?

**Comment by mshrtsr** [comment]
@tuanphamcybozu 
Sorry, I want you to ..

## yamory.yml

- Please remove `cache: pnpm` from `actions/setup-node`

## license.yml

- Please use `pnpm/actions-setup` 
- Please add `cache: pnpm` option to `actions/setup-node`
- No `actions/cache@v3` is necessary

## Other workflows

Please keep it as is as the original of `main` branch.
(using setup-node with `cache` option, no `actions/cache` step)

**Comment by tuanphamcybozu** [comment]
@mshrtsr 
I updated to original cache in other jobs.
Thnks for careful review 🎉

**Review by tasshi-me — APPROVED** [review]
LGTM!!
